packages/frontend/src/components/composer/Composer.tsx

```diff
@@ -974,6 +974,70 @@ export function useDraft(
   )
 
   const { jumpToMessage } = useMessage()
+
+  // Cache for message IDs to avoid expensive RPC calls
+  const quotableMessageIdsCache = useRef<{
+    chatId: number | null
+    messageIds: number[] | null
+  }>({ chatId: null, messageIds: null })
+
+  const getCachedQuotableMessageIds = useCallback(async (): Promise<
+    number[]
+  > => {
+    if (chatId === null) return []
+
+    // Check if we have cached data for the current chat
+    if (
+      quotableMessageIdsCache.current.chatId === chatId &&
+      quotableMessageIdsCache.current.messageIds
+    ) {
+      return quotableMessageIdsCache.current.messageIds
+    }
+
+    // Fetch fresh data and cache it
+    const messageIds = await BackendRemote.rpc.getMessageIds(
+      accountId,
+      chatId,
+      false,
+      false
+    )
+
+    const infoMessages = await BackendRemote.rpc.getMessageListItems(
+      accountId,
+      chatId,
+      true, // only info messages
+      false //no daymarkers
+    )
+    const infoMessageIds = new Set(
+      infoMessages
+        .filter(msg => msg.kind === 'message' && 'msg_id' in msg)
+        .map(msg => (msg as { msg_id: number }).msg_id)
+    )
+
+    // Filter out info messages
+    const filteredMessageIds = messageIds.filter(id => !infoMessageIds.has(id))
+
+    quotableMessageIdsCache.current = {
+      chatId,
+      messageIds: filteredMessageIds,
+    }
+
+    return filteredMessageIds
+  }, [accountId, chatId])
+
+  // Invalidate message IDs cache when the chat list changes
+  useEffect(() => {
+    if (chatId === null) return
+    const cleanup = onDCEvent(accountId, 'IncomingMsg', args => {
```

Comment:
When you send a message yourself, the list does not update (same with deletion, as you mentioned).